WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostListFragment.java

```diff
@@ -998,9 +999,46 @@ public void onDataLoaded(boolean isEmpty) {
                                             .add(ReaderPostWebViewCachingFragment.newInstance(blogId, postId), tag)
                                             .commit();
                     }
+
+                    // show snackbar when saving post not in saved posts list
+                    if (isBookmarked && getPostListType() == ReaderPostListType.TAG_FOLLOWED
+                        && (mCurrentTag != null && mCurrentTag.isBookmarked())) {
+                        showBookmarkSnackbar();
+                    }
                 }
             };
 
+    private void showBookmarkSnackbar() {
+        if (!isAdded()) {
+            return;
+        }
+
+        Snackbar.make(getView(), R.string.reader_bookmark_snack_title,
+                AccessibilityUtils.getSnackbarDuration(getActivity())).setAction(R.string.reader_bookmark_snack_btn,
+                new View.OnClickListener() {
+                    @Override public void onClick(View view) {
+                        // if the fragment is part of WPMainActivity we just need to update filter and scroll to
+                        // the toolbar. This implementation assumes that Reader list in WPMainActivity will have a
+                        // toolbar.
+                        if (getActivity() instanceof WPMainActivity) {
+                            if (ReaderTagTable.getBookmarkTags().isEmpty()) { // sanity check
+                                return;
+                            }
+                            mRecyclerView.setCurrentFilter(ReaderTagTable.getBookmarkTags().get(0));
+                            mRecyclerView.showToolbar();
+                        } else {
+                            // in case of independent reader list, go back to WPMainActivity
+                            // and instruct it to show saved posts in Reader Tab
+                            Intent intent = new Intent(view.getContext(), WPMainActivity.class);
```

Comment:
One more comment - shouldn't we put this code into ActivityLauncher so we can reuse it in both PostsList and PostDetail? Wdyt?

Reply:
👍